src/main/java/com/weirddev/testme/intellij/ui/classconfigdialog/CreateTestMeDialog.java

```diff
@@ -0,0 +1,197 @@
+        init();
+    }
+
+    @Override
+    protected JComponent createCenterPanel() {
+        JPanel panel = new JPanel();
+        panel.setLayout(new BoxLayout(panel, BoxLayout.Y_AXIS));
+
+        JLabel fieldLabel = new JLabel("Select Need Mocked Fields");
+        panel.add(fieldLabel);
+        panel.add(ScrollPaneFactory.createScrollPane(myFieldsTable));
+
+        JLabel methodLabel = new JLabel("Select Need Test Methods");
+        panel.add(methodLabel);
+        panel.add(ScrollPaneFactory.createScrollPane(myMethodsTable));
+
+        initExtractingClassMembers();
+        return panel;
+    }
+
+    /**
+     * update field is mockable or not after user checked
+     */
+    private void updateClassMockableFields() {
+        Collection<MemberInfo> selectedMemberInfos = myFieldsTable.getSelectedMemberInfos();
+        List<String> userCheckedMockableFieldsList =
+            selectedMemberInfos.stream().map(e -> e.getMember().getName()).toList();
+        Type classTypeObj = (Type)templateCtxtParams.get(TestMeTemplateParams.TESTED_CLASS);
+        for (Field field : classTypeObj.getFields()) {
+            if (userCheckedMockableFieldsList.contains(field.getName())) {
+                checkedFieldNameList.add(field.getName());
+            }
+        }
+        templateCtxtParams.put(TestMeTemplateParams.USER_CHECKED_MOCK_FIELDS, checkedFieldNameList);
+    }
+
+    /**
+     * update method is testable or not after user checked
+     */
+    private void updateClassTestableMethods() {
+        Collection<MemberInfo> selectedMemberInfos = myMethodsTable.getSelectedMemberInfos();
+        List<String> testableMethodList =
+            selectedMemberInfos.stream().map(e -> PsiMethodUtils.formatMethodId((PsiMethod)e.getMember())).toList();
+        Type classTypeObj = (Type)templateCtxtParams.get(TestMeTemplateParams.TESTED_CLASS);
+        for (Method method : classTypeObj.getMethods()) {
+            if (testableMethodList.contains(method.getMethodId())) {
+                checkedMethodIdList.add(method.getMethodId());
+            }
+        }
+        templateCtxtParams.put(TestMeTemplateParams.USER_CHECKED_TEST_METHODS, checkedMethodIdList);
+    }
+
+    /**
+     * init and extract class fields and methods for user to check
+     */
+    public void initExtractingClassMembers() {
+        Set<PsiClass> classes= InheritanceUtil.getSuperClasses(myTargetClass);
+        classes.add(myTargetClass);
+
+        Type classTypeObj = (Type)templateCtxtParams.get(TestMeTemplateParams.TESTED_CLASS);
+        TestSubjectInspector testSubjectUtil =
+            (TestSubjectInspector)templateCtxtParams.get(TestMeTemplateParams.TestSubjectUtils);
+        MockBuilder templateMockBuilder = getTemplateMockBuilder(templateFileName);
+
+        // build field mockable map, key = field name, value = true - if field mockable
+        Map<String, Boolean> fieldMockableMap = new HashMap<>();
+        for (Field field : classTypeObj.getFields()) {
```

Comment:
implementation is relying on TestTemplateContextBuilder.build() being called - this method will introspect and initialize test subject object graph recursively. it's a very expensive action. it can take a long time, depending on tested code complexity and it's dependencies. it will result in bad UX if user has to wait long between 2 dialog actions in order to complete user input for test generation. I realize the implementation complexities, please try to find a way to initialize the list of relevant methods/fields without relying on typeDictionary initialization at this stage